hls/decode-util_test.go

```diff
@@ -250,8 +249,8 @@ func TestReadMediaPlaylist(t *testing.T) {
 		if s.URI != newP.Segments[i].URI {
 			t.Errorf("Expected URI to be %s, but got %s", s.URI, newP.Segments[i].URI)
 		}
-		if s.Map != nil && !reflect.DeepEqual(s.Map, newP.Segments[i].Map) {
-			t.Errorf("Expected %d Segment Map tp be %v, but got %v", i, s.Map, newP.Segments[i].Map)
+		if s.Map != nil && s.Map.URI != newP.Segments[i].Map.URI {
```

Comment:
Yeah, this is a pita, check out https://github.com/google/go-cmp if you haven't seen it already.

Specifically:
```
equality is determined by recursively comparing the primitive kinds on both values,
much like reflect.DeepEqual. Unlike reflect.DeepEqual, unexported fields are not 
compared; they result in panics unless suppressed by using an Ignore option.
```

I understand the desire to not include external dependencies, so it's just a suggestion.